src/Tests/Receiving/MessageExtensionsTests.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus.Tests.Receiving
+{
+    using System;
+    using Configuration;
+    using Microsoft.Azure.ServiceBus;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class MessageExtensionsTests
+    {
+        [TestCase]
+        public void Should_not_fail_to_process_octectstream_message_without_body()
+        {
+            var message = new Message();
+            var body = message.GetBody();
+            Assert.AreEqual(Array.Empty<byte>(), body);
+        }
+
+        [TestCase]
+        public void Should_not_fail_to_process_bytearray_message_without_body()
+        {
+            var message = new Message();
+            message.UserProperties[TransportMessageHeaders.TransportEncoding] = "wcf/byte-array";
+            var body = message.GetBody();
+            Assert.AreEqual(Array.Empty<byte>(), body);
```

Comment:
I'd prefer newlines between AAA phases

Reply:
I'm 👍 to separate the message and it's assignments from body and it's test. Test though is coupled to the body which is a system under test.